app/services/word_watcher.rb

```diff
@@ -55,12 +65,34 @@ def should_flag?
   end
 
   def should_block?
-    word_matches_for_action?(:block)
+    word_matches_for_action?(:block, all_matches: true)
   end
 
-  def word_matches_for_action?(action)
-    r = self.class.word_matcher_regexp(action)
-    r ? r.match(@raw) : false
-  end
+  def word_matches_for_action?(action, all_matches: false)
+    regexp = self.class.word_matcher_regexp(action)
+    if regexp
+      match = regexp.match(@raw)
+      return match if !all_matches || !match
 
+      matches = []
```

Comment:
It would be better to make this a `Set.new` which ensures it's unique as you insert.

Reply:
Done ✅